### PR TITLE
feat: add additional option to control the default_conn_delay option of the limit-conn plugin

### PR DESCRIPTION
**Pull request by zuiyangqingzhou**
### What this PR does / why we need it:
feat: [issues](https://github.com/apache/apisix/issues/4583)

ps:  

I'm thinking how to do the unit test here

Because I cannot verify the logic of the third-party library [lua-resty-limit-traffic](https://github.com/openresty/lua-resty-limit-traffic/blob/master/lib/resty/limit/conn.lua#L89-L104), I can only verify whether nil is passed for  the req_latency parameter, but what should I do with this unit test?

Can you give some advice?

### CODE REVIEW

File: docs/en/latest/plugins/limit-conn.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Limiting request concurrency plugin.
...
+| delay_strict_mode  | boolean | optional    | false   | [true,false]                                                                              | enable the strict mode of the latency seconds. If you set this option to `true`, it will run strictly according to the latency seconds you set without additional calculation logic.                                                                                                                                                                                                                                                                                                                      |
```
Comment (tokers) [inline docs/en/latest/plugins/limit-conn.md:43]: I think the option name is not so straightforward but I didn't come out with a more proper one :(

Comment (zuiyangqingzhou) [reply docs/en/latest/plugins/limit-conn.md:43]: 😂

Comment (spacewander) [reply docs/en/latest/plugins/limit-conn.md:43]: What about `only_use_default_delay`?

Comment (zuiyangqingzhou) [reply docs/en/latest/plugins/limit-conn.md:43]: `only_use_default_delay`,   `use_default_conn_delay`,  `apply_default_conn_delay`,  `stint_default_conn_delay`, Which of the four do you think is better, or do you have better suggestions?

Comment (spacewander) [reply docs/en/latest/plugins/limit-conn.md:43]: I vote for the first one. What about others' opinions?

Comment (zuiyangqingzhou) [reply docs/en/latest/plugins/limit-conn.md:43]: @tokers  what do you think?

Comment (tokers) [reply docs/en/latest/plugins/limit-conn.md:43]: Vote for the first one too, it's easy to understand.

Comment (zuiyangqingzhou) [reply docs/en/latest/plugins/limit-conn.md:43]: Okay, I will change it to this one. 
What about the test case I mentioned above? Any suggestions? @spacewander @tokers

Comment (spacewander) [reply docs/en/latest/plugins/limit-conn.md:43]: You can take a look at the test in https://github.com/apache/apisix/pull/4585

File: docs/zh/latest/plugins/limit-conn.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ title: limit-conn
...
-| delay_strict_mode  | boolean | optional | false  | [true,false]                                                                              | 延迟时间的严格模式. 如果设置为`true`的话，将会严格按照设置的时间来进行延迟                                                                                                                                                                                                                                                                                                                                                                           |
+| only_use_default_delay  | boolean | optional | false  | [true,false]                                                                              | 延迟时间的严格模式. 如果设置为`true`的话，将会严格按照设置的时间来进行延迟                                                                                                                                                                                                                                                                                                                                                                           |
```
Comment (tokers) [inline docs/zh/latest/plugins/limit-conn.md:33]: `.` is the period symbol in English, shall we use `。`?

File: t/plugin/limit-conn2.t [hunk 3]
```diff
@@ -178,3 +178,76 @@ qr/request latency is/
...
+=== TEST 6: hit route
+--- log_level: debug
+--- request
+GET /hello1
+--- grep_error_log eval
+qr/request latency is/
+--- grep_error_log_out
+request latency is
+request latency is
+
```
Comment (tokers) [inline t/plugin/limit-conn2.t:254]: Redundant empty line.

Comment (zuiyangqingzhou) [reply t/plugin/limit-conn2.t]: ok, i will fix it

**Review by tokers — APPROVED** [review]

File: t/plugin/limit-conn2.t [hunk 4]
```diff
@@ -178,3 +178,76 @@ qr/request latency is/
...
+passed
+
+
+
+=== TEST 6: hit route
+--- log_level: debug
+--- request
+GET /hello1
+--- grep_error_log eval
+qr/request latency is/
```
Comment (spacewander) [inline t/plugin/limit-conn2.t:250]: Need to check latency in this mode is `nil`.

Comment (zuiyangqingzhou) [reply t/plugin/limit-conn2.t:250]: I don't know exactly what you mean, 

> https://github.com/apache/apisix/pull/4604/files/47dc32979a94977aa8e3d0e73270dc88a9c43149#diff-2f0a07663e8e90e9f64519ee946aaf82729c8bb43e4f9100902e87d0a325f78eR251-R253

hasn't grep_error_log_out already check that the latency is nil? 
Or would you give some guidance on how to do it ?

Comment (spacewander) [reply t/plugin/limit-conn2.t:250]: The grep pattern should be `request latency is nil`.

File: t/plugin/limit-conn2.t [hunk 5]
```diff
@@ -178,3 +178,76 @@ qr/request latency is/
...
+--- grep_error_log_out
+request latency is
```
Comment (tzssangglass) [inline t/plugin/limit-conn2.t:252]:
```suggestion
request latency is nil
```

**Review by spacewander — APPROVED** [review]

**Pull request merged**